### PR TITLE
use custom Function interface instead of java 8

### LINKED ISSUE #1917: Make status-go upgrades possible without touching react-native-webview-bridge
### User Story

As a developer, I don't want to have to do go to https://github.com/status-im/react-native-webview-bridge and commit an update status-go commit when I should be able to do it directly from status-react. It's also easy to forget and leads to annoying bugs.

### Description

*Type*: Bug/Tooling

### Solution
*Summary*: Turn these two PRs:

- https://github.com/status-im/status-react/pull/1892/files
- https://github.com/status-im/react-native-webview-bridge/commit/354ec944d5f611bc41db5aee94dc795d4b635b0e

into one.

Or, even better, Specify version as ENV variable in a single place.

### PR DESCRIPTION
**Pull request by rasom**
another attempt to fix #1917 

status: ready

### CODE REVIEW

File: modules/react-native-status/android/build.gradle [hunk 1]
```diff
@@ -15,5 +15,6 @@ android {
 dependencies {
     compile 'com.facebook.react:react-native:+'
     compile 'com.instabug.library:instabug:3+'
+    compile 'status-im:function:0.0.1'
```
Comment (oskarth) [inline modules/react-native-status/android/build.gradle:18]: What is this?

File: android/app/src/main/java/im/status/ethereum/MainApplication.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
-import java.util.function.Function;
+import im.status.ethereum.function.Function;
```
Comment (oskarth) [inline android/app/src/main/java/im/status/ethereum/MainApplication.java:35]: I don't see where this is

Comment (rasom) [reply android/app/src/main/java/im/status/ethereum/MainApplication.java:35]: this lives in 
`compile 'status-im:function:0.0.1' `
ant this is maven project which literally contains only
```java
package im.status.ethereum.function;

public interface Function<T, R> {
    R apply(T var1);
}
```
i would prefer to duplicate this piece of code in all projects, but gradle will not build it...

**Review by flexsurfer — APPROVED** [review]

**Review by yenda — APPROVED** [review]

**Review by oskarth — APPROVED** [review]

**Review by janherich — APPROVED** [review]

File: modules/react-native-status/android/build.gradle (hunk 1 above)
Comment (oskarth) [reply modules/react-native-status/android/build.gradle:18]: EDIT: Custom mini maven package: https://github.com/status-im/function

**Pull request merged**